### ISSUE TITLE
==⚠  Switch to EXCALIDRAW VIEW in the MORE OPTIONS menu of this document. ⚠==

### ISSUE THREAD

**Issue by subjektivdk** [zsviczian/obsidian-excalidraw-plugin#511]
Windows 10 Pro
Obsidian v 0.13.30
Excalidraw v 1.6.17
----------------------------------
Get this error now and can't edit drawings. PNG are rendered. 

When clicking "Open as Excalidraw drawing" this error shows. I have tried with both compressed/uncompressed json.
![2022-03-08 11_36_09-Window](https://user-images.githubusercontent.com/10208704/157219874-c9aa3a4c-b9cd-4138-a8f8-74f6537282a9.png)

**Comment by zsviczian** [comment]
does this effect all your drawing, or just this one? Seems something has happened to the file. can you share this file so I can take a look?

**Comment by zsviczian** [comment]
> I was just coming here to file an issue about this exact message, except on publish. I was going to ask if it's possible to have the message mention something like "this is best viewed locally" or something that makes sense on publish? I like the way the publish version has all the text and links (very impressive!) and don't want to _just_ export it to svg.

I don’t understand. You are exporting your vault via publish including Excalidraw drawings, and when users open the exported version this error comes up? Or this comes up during export? or? A screenshot, a screen recording, or maybe a link where I can check to see would be helpful.

Are you having problem saving/opening your drawings in general, or this is just connected with Publish?

**Comment by eleanorkonik** [comment]
I sent you a DM on Discord since some of the info is private, but for me this is just connected to publish. Sorry to hijack another issue I just wasn't sure ... what else to call my issue since it's the same error message 😅

**Comment by subjektivdk** [comment]
It happens on all my drawings

**Comment by subjektivdk** [comment]
I can't figure out how to repair the file. So I can't edit the drawing afterwards

**Comment by zsviczian** [comment]
This file looks like the application was terminated during saving. The file is simply cut in half.
How do you close drawings? Do you terminate Obsidian, or do you navigate to another file in the vault?
Since when is this happening?
Do you use Obsidian sync or any other version management like git, etc.? With Obsidian sync you can search for the last working version of the file.

![image](https://user-images.githubusercontent.com/14358394/157397647-e787237c-6589-4d5b-9f5c-df266ceada4d.png)

This is what I was able to recover from the file. Not much, I'm afraid...
[udfakturering.2022-03-08.09.22.44.excalidraw.md](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8212816/udfakturering.2022-03-08.09.22.44.excalidraw.md)

**Comment by subjektivdk** [comment]
> 

I use Obsidian sync, and all files are backed up on my Synology NAS from my desktop pc. 
I do not terminate Obsidian. I might navigate to another file, but I am also old school and click the save button pretty consistent.
I suspect the two-way synchronization to the NAS to be the problem. I have just made a change so that the Obsidian vault are only beeing backed up (one-way)

**Comment by subjektivdk** [comment]
> This file looks like the application was terminated during saving. The file is simply cut in half. How do you close drawings? Do you terminate Obsidian, or do you navigate to another file in the vault? Since when is this happening? Do you use Obsidian sync or any other version management like git, etc.? With Obsidian sync you can search for the last working version of the file.
> 
> ![image](https://user-images.githubusercontent.com/14358394/157397647-e787237c-6589-4d5b-9f5c-df266ceada4d.png)
> 
> This is what I was able to recover from the file. Not much, I'm afraid... [udfakturering.2022-03-08.09.22.44.excalidraw.md](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8212816/udfakturering.2022-03-08.09.22.44.excalidraw.md)

Thanks for trying to recover the drawing.

**Comment by zsviczian** [comment]
I am also an old school guy. Click save. Maintain multiple backups. Create versions of the same file with naming a convention that reflects version and date... the devil never sleeps.

Saving has been a pain with Obsidian, because Excalidraw files are much less resilient than plain markdown files. Also Excalidraw files are relatively large. I tried creating a backup before the save operation... but that caused other issues with sync...

If you find out the root cause, please share so I can learn from it and see what more I can do.

**Comment by subjektivdk** [comment]
> I am also an old school guy. Click save. Maintain multiple backups. Create versions of the same file with naming a convention that reflects version and date... the devil never sleeps.
> 
> Saving has been a pain with Obsidian, because Excalidraw files are much less resilient than plain markdown files. Also Excalidraw files are relatively large. I tried creating a backup before the save operation... but that caused other issues with sync...
> 
> If you find out the root cause, please share so I can learn from it and see what more I can do.

Sure, will do

**Comment by zsviczian** [comment]
Alt+Ctrl+left arrow key is configured for you as the hotkey for "Navigate back", or?

**Comment by zsviczian** [comment]
ok, so Alt+Ctrl+Left for you is configured as "Focus on pane to the left", correct?
Do you switch panes while in fullscreen mode... how? Are you using a different plugin for fullscreen mode?

**Comment by zsviczian** [comment]
@Yot67, can you try to replace your main.js in the `.obsidian/plugins/obisidian-excalidraw-plugin` folder with this one, and then let me know if you can open the file?
[main.zip](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8225923/main.zip). You will need to unzip to find the main.js.

**Comment by zsviczian** [comment]
if this new main.js does not solve your problem, can I ask you to share the broken file with me so I can take a look what has happened?

**Comment by Yot67** [comment]
I'm using those for navigate only :
![Obsidian_Jf28ATAfNI](https://user-images.githubusercontent.com/14005233/157726368-d4ea2806-c988-40fd-9726-2dd337c8efee.png)
Sorry for the wrong information I don't use fulscreen at all. I just use one TAB in a plain window.

I can reproduce the "bug" every time I navigate back (Ctrl+Alt+<-) from the .excalidraw to the original TAB.
At a point where I've lost not only the concerns .excalidraw but it erased completely the original .md and some parts of others .md. 
This bug not only affect .excalidraw.

**Comment by Yot67** [comment]
> @Yot67, can you try to replace your main.js in the `.obsidian/plugins/obisidian-excalidraw-plugin` folder with this one, and then let me know if you can open the file? [main.zip](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8225923/main.zip). You will need to unzip to find the main.js.

he deleted the .md file, I've lost all my work : 0ko

**Comment by zsviczian** [comment]
> At a point where I've lost not only the concerns .excalidraw but it erased completely the original .md and some parts of others .md.
This bug not only affect .excalidraw.

This sounds like the issue is not caused by Excalidraw, but some other plugin.  Can you disable other plugins, restart Obsidian and test, just to be sure?

**Comment by Yot67** [comment]
I've turned off all the community plugins except Excalidraw. And I've made 3 excalidraw from one .md, and it did it again. 
![Obsidian_flNSuY1kfn](https://user-images.githubusercontent.com/14005233/157741869-d3b80292-889d-414d-bfc7-937d30d9997e.png)
`---

excalidraw-plugin: parsed

---
==⚠  Switch to EXCALIDRAW VIEW in the MORE OPTIONS menu of this document. ⚠==


# Text Elements
tsetstetsqteqsetqsetqstqset ^Vx6xJTrn

%%
# Drawing
```json
{
	"type": "excalidraw",
	"version": 2,
	"source": "https://excalidraw.com",
	"elements": [
		{
			"id": "Vx6xJTrn",
			"type": "text",
			"x": -321.26666259765625,
			"y": -216.72379684448242,
			"width": 299,
			"height": 25,
			"angle": 0,
			"strokeColor": "#000000",
			"backgroundColor": "transparent",
			"fillStyle": "hachure",
			"strokeWidth": 1,
			"strokeStyle": "solid",
			"roughness": 1,
			"opacity": 100,
			"groupIds": [],
			"strokeSharpness": "sharp",
			"seed": 866694455,
			"version": 29,
			"versionNonce": 1458206967,
			"isDeleted": false,
			"boundElements": null,
			"updated": 1646933792723,
			"link": null,
			"text": "tsetstetsqteqsetqsetqstqset",
			"rawText": "tsetstetsqteqsetqsetqstqset",
			"fontSize": 20,
			"fontFamily": 1,
			"textAlign": "left",
			"verticalAlign": "top",
			"baseline": 18,
			"containerId": null,
			"originalText": "tsetstetsqteqsetqsetqstqset"
		}
	],
	"appState": {
		"theme": "dark",
		"viewBackgroundColor": "#ffffff",
		"currentItemStrokeColor": "#000000",
		"currentItemBackgroundColor": "transparent",
		"currentItemFillStyle": "hachure",
		"currentItemStrokeWidth": 1,
		"currentItemStrokeStyle": "solid",
		"currentItemRoughness": 1,
		"currentItemOpacity": 100,
		"currentItemFontFamily": 1,
		"currentItemFontSize": 20,
		"currentItemTextAlign": "left",
		"currentItemStrokeSharpness": "sharp",
		"currentItemStartArrowhead": null,
		"currentItemEndArrowhead": "arrow",
		"currentItemLinearStrokeSharpness": "round",
		"gridSize": null,
		"colorPalette": {}
	},
	"files": {}
}
```
%%
[Tuto Dev Web.md.txt](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8226766/Tuto.Dev.Web.md.txt)
`

**Comment by zsviczian** [comment]
hm... did you restart Obsidian before doing so?
Do you have any file synchronization going on in the background? like onedrive, google drive, or similar?

**Comment by zsviczian** [comment]
This file opens for me without any issues... I don't understand. Is this file not opening for you as excalidraw?
[Tuto.Dev.Web.md](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8226793/Tuto.Dev.Web.md)

**Comment by Yot67** [comment]
[Tuto.Dev.Web.md](https://github.com/zsviczian/obsidian-excalidraw-plugin/files/8226793/Tuto.Dev.Web.md) is the initial file on wich I worked on when the matter occured first time.
I don't use file synchronizaion.
I 've just restarted with only Excalidraw as Community Plugin and I can't reproduce the bug anymore.

**Comment by zsviczian** [comment]
I am afraid that you have another plugin causing this issue. If a plugin is poorly written, it will not completely stop, even if you stop it in community plugins. You need to restart Obsidian.

Now the question is, which plugin was causing the issue.

Can you list the plugins you have installe?

**Comment by Yot67** [comment]
console.log(Object.keys(app.plugins.manifests).join("\n"))
VM129:1 calendar
customizable-page-header-buttons
dataview
find-and-replace-in-selection
image-window
obsidian-checklist-plugin
obsidian-excalidraw-plugin
obsidian-icon-folder
obsidian-markdown-formatting-assistant-plugin
oz-image-plugin
table-editor-obsidian
tag-wrangler
templater-obsidian
wikilinks-to-mdlinks-obsidian
zoottelkeeper-obsidian-plugin

![Obsidian_DZIDPsYWkF](https://user-images.githubusercontent.com/14005233/157745682-ae757d36-e273-4af1-a6cf-02274eda8b84.png)

**Comment by zsviczian** [comment]
From your list plugins I also have enabled in my vault are dataview, templater, customizable-page-header. I am pretty confident those are not causing your issues.

Since live preview, I think you don't need zo-image-plugin... but I might be mistaken.

What you can do is to turn them on one by one, until you can reproduce the error again.

**Comment by Yot67** [comment]
I can't reproduce the bug anymore.

**Comment by zsviczian** [comment]
even if you turn all the plugins on again?

**Comment by Yot67** [comment]
yes

**Comment by zsviczian** [comment]
let me know if it happens again... though I really hope it won't.
If it does, then please also send a screenshot of the Console tab of Developer Console.

**Comment by Yot67** [comment]
I'll do, and this time I will automate backups. Thank you for your help.

**Comment by subjektivdk** [comment]
> I am also an old school guy. Click save. Maintain multiple backups. Create versions of the same file with naming a convention that reflects version and date... the devil never sleeps.
> 
> Saving has been a pain with Obsidian, because Excalidraw files are much less resilient than plain markdown files. Also Excalidraw files are relatively large. I tried creating a backup before the save operation... but that caused other issues with sync...
> 
> If you find out the root cause, please share so I can learn from it and see what more I can do.

So I am pretty sure it was the Synology Drive sync in conjunction with Obsidian sync that messed it up. The Last week I have only been using one-way backup to my Synology NAS, an the problem has not occured.

**Comment by zsviczian** [comment]
I implemented a number of additional safeguards in 1.6.21. Hopefully similar incidents will not happen. I am closing this issue. Please open a new one in case you encounter problems.

**Comment by kevinkoosk** [comment]
I used a hotkey to toggle the Markdown vs Excalidraw modes. That helped me to get the page into Excalidraw mode.

![image](https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/7147578/6998c080-24c0-4d44-a24f-ca53d3a82a39)
